code/script.js

```diff
@@ -220,106 +227,177 @@ const generateBoard = () => {
+};
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  generateBoard();
+  setSecret();
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.options[questions.selectedIndex].value;
+
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
+const checkQuestion = (currentQuestion) => {
+  selectQuestion()
+  const { category, value } = currentQuestion;
+  const matchingCharacters = [];
+  for (const character of charactersInPlay) {
+    let shouldKeep = false;
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
 
-  } else if (category === 'accessories' || category === 'other') {
+    // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+    // See if we should keep or remove people based on that
+    // Then invoke filterCharacters
+    if ((category === 'hair' || category === 'eyes') && character[category] === valye) {
+      shouldKeep = true;
 
-  }
-}
+    } else if ((category === 'accessories' || category === 'other') && character[category].includes(value)) {
+      shouldKeep = true;
+    }
+
+
+    if ((keep && shouldKeep) || (!keep && !shouldKeep)) {
 
-// It'll filter the characters array and redraw the game board.
-const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      matchingCharacters.push(character);
     }
-  } else if (category === 'other') {
-    // Similar to the one above
+    filterCharacters(keep, matchingCharacters);
+
+  };
+
+  const filterCharacters = (keep, matchingCharacters) => {
```

Comment:
There are two filterCharacter functions, when I plug your code into VScode it shows several functions that don't have closing curly brackets. Both filterCharacter functions and checkQuestion function.